system/lib/wasmfs/syscalls.cpp

```diff
@@ -1014,8 +1014,8 @@ int __syscall_renameat(int olddirfd,
   }
 
   // Perform the move.
-  if (!lockedNewParent.insertMove(newFileName, oldFile)) {
-    return -EPERM;
+  if (auto err = lockedNewParent.insertMove(newFileName, oldFile)) {
+    return err;
```

Comment:
Maybe in places like this it would be good to `assert(err < 0)`? (otherwise we are just checking `!= 0`). One other place above too.

Reply:
Good idea 👍

Reply:
Done separately in https://github.com/emscripten-core/emscripten/pull/17786